phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java

```diff
@@ -312,7 +312,10 @@ public Job getJob() throws Exception {
                     return configureJobForAysncIndex();
                 }
                 else {
-                    //Local and non-transactional global indexes to be built on the server side
+                    // Local and non-transactional global indexes to be built on the server side
+                    // It is safe to unset CURRENT_SCN_VALUE for server side rebuilds to make sure that
+                    // all the rows that has been inserted so far will be rebuilt
+                    configuration.unset(PhoenixConfigurationUtil.CURRENT_SCN_VALUE);
```

Comment:
IMO, this configuration change should be inside configureJobForServerBuildIndex() so that any future calls to it do not miss it. I noticed that configureJobForServerBuildIndex() configuration.set() calls, so this would not be out of place.

Reply:
Instead of moving this to configureJobForServerBuildIndex(), I restructured the code in this method to convince you that this is the right place to set or not to set SCN. Also, please consider that we may want to introduce the SCN option in future. In that case, it should be handled here.